### PR TITLE
Update hydrosphere examples

### PR DESCRIPTION
**Pull request by BeardedWhale**

### CODE REVIEW

**Comment by Valenzione** [comment]
https://github.com/Hydrospheredata/hydro-serving-example/blob/Update_Liza_Nikita/examples/adult/demo/Adult_demo.ipynb seems to be a binarizer demo notebook, not an adult one

**Comment by Valenzione** [comment]
Minor comment - https://github.com/Hydrospheredata/hydro-serving-example/blob/Update_Liza_Nikita/examples/face_recognition/demo/Face_Detection_Demo.ipynb has a redundant cell with `from time import sleep` statement. `sleep` is never used in this notebook later

**Comment by Valenzione** [comment]
https://github.com/Hydrospheredata/hydro-serving-example/blob/Update_Liza_Nikita/examples/mobilenet/demo/Mobile_net_demo.ipynb - missing image output from `img.show()` cell

**Comment by Valenzione** [comment]
https://github.com/Hydrospheredata/hydro-serving-example/blob/Update_Liza_Nikita/examples/fraud_detection/demo/fraud_demo.ipynb

Why do we need `%%time` magic while reading data?

**Comment by Valenzione** [comment]
https://github.com/Hydrospheredata/hydro-serving-example/blob/Update_Liza_Nikita/examples/python_mnist/grpc_request.py use `np.ones` with 28*28 shape instead of inlining this 28*28 array of ones in code.

**Comment by Valenzione** [comment]
https://github.com/Hydrospheredata/hydro-serving-example/blob/Update_Liza_Nikita/examples/python_mnist/demo/Mnist_demo.ipynb use `%%capture` magic during import to capture warnings

Why output tensor field `classes` contains 10 `int_vals` for one sample? It should contain 1 class label and 10 probabilities for each sample provided.

**Comment by Valenzione** [comment]
IMHO rename to `tf_mnist` https://github.com/Hydrospheredata/hydro-serving-example/tree/Update_Liza_Nikita/examples/mnist

**Comment by Valenzione** [comment]
https://github.com/Hydrospheredata/hydro-serving-example/blob/Update_Liza_Nikita/examples/mnist/basic-api.py should be called `train_mnist.py`

**Comment by Valenzione** [comment]
https://github.com/Hydrospheredata/hydro-serving-example/blob/Update_Liza_Nikita/examples/activity_recognition/demo/AR_demo.ipynb
Change header name, remove commented prints, remove empty cells. Reshape and decode output, so it will be understandable which class was returned

**Comment by Valenzione** [comment]
User shorter and more concise headers, without `How to`. Instead of `How to train model` - `Training`, `How to deploy` - `Deployment` etc.

**Comment by Valenzione** [comment]
```python
import grpc
from grpc import ssl_channel_credentials
```
Why use two lines instead of one?

**Comment by Valenzione** [comment]
Remove `signature_name` from `ModelSpec` instantiation

File: examples/mobilenet/model/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+numpy==1.16.2
+keras
```
Comment (Valenzione) [inline examples/mobilenet/model/requirements.txt]: Fix keras version. After recent update, package `mobilenetv2` was renamed to `mobilenet_v2`. I suggest using `keras==2.2.5` and changing mobilenet example code from `mobilenetv2` to `mobilenet_v2`

File: examples/python_mnist/grpc_request.py [hunk 2]
```diff
@@ -0,0 +1,860 @@
...
+import hydro_serving_grpc as hs
+from grpc import ssl_channel_credentials
+
+#            import hydro_serving_grpc as hs 
+
+channel = grpc.secure_channel("dev.k8s.hydrosphere.io", credentials=ssl_channel_credentials())
+stub = hs.PredictionServiceStub(channel) 
+model_spec = hs.ModelSpec(name="mnist2", signature_name="INFERRED_PIPELINE_SIGNATURE")
+tensor_shape = hs.TensorShapeProto(dim=[hs.TensorShapeProto.Dim(size=-1), hs.TensorShapeProto.Dim(size=28), hs.TensorShapeProto.Dim(size=28)])
+val = np.array([
```
Comment (Valenzione) [inline examples/python_mnist/grpc_request.py]: Use `val = np.ones((1, 28, 28), dtype=np.float64))` instead of literal. Or load example MNIST image`

File: examples/python_mnist/model/serving.yaml [hunk 3]
```diff
@@ -0,0 +1,24 @@
+kind: Model
+name: "mnist2"
```
Comment (Valenzione) [inline examples/python_mnist/model/serving.yaml]: Why `mnist2` and not `py_mnist`?

File: examples/python_mnist/model/requirements.txt [hunk 4]
```diff
@@ -0,0 +1,4 @@
+scikit-learn==0.20.2
+numpy==1.16.2
+keras
```
Comment (Valenzione) [inline examples/python_mnist/model/requirements.txt]: Fix Keras and TensorFlow versions, so this demo won't break after TF 2.0 release. This can happen since TF 2.0 is not backwards compatible with 1.0

**Comment by Valenzione** [comment]
image
@GurovNik Why cell output contradicts code written inside cell block? In [this](https://github.com/Hydrospheredata/hydro-serving-example/blob/Update_Liza_Nikita/examples/activity_recognition/demo/activity_recognition_demo.ipynb) notebook

**Comment by Valenzione** [comment]
Also no output in [this](https://github.com/Hydrospheredata/hydro-serving-example/blob/Update_Liza_Nikita/examples/binarizer/demo/binarizer_demo.ipynb) notebook. Also no printed tensor.

**Comment by GurovNik** [comment]
What mean contradicts the code?

On 3 Sep 2019, at 16:29, Yuriy <notifications@github.com<mailto:notifications@github.com>> wrote:


[image]<https://user-images.githubusercontent.com/15328647/64169714-0e5d8f80-ce57-11e9-9044-8cf889e1f7ab.png>
@GurovNik<https://github.com/GurovNik> Why cell output contradicts code written inside cell block? In this<https://github.com/Hydrospheredata/hydro-serving-example/blob/Update_Liza_Nikita/examples/activity_recognition/demo/activity_recognition_demo.ipynb> notebook

**Comment by Valenzione** [comment]
Why do we need `random_forest` directory [here](https://github.com/Hydrospheredata/hydro-serving-example/tree/Update_Liza_Nikita/examples/fraud_detection/model)? it seems like a duplicate of the model pickle. Delete it?

**Comment by GurovNik** [comment]
why adult demo address "dev.k8s.hydrosphere.io" again?

File: README.md [hunk 5]
```diff
@@ -1,24 +1,54 @@
...
- ```bash
- pip install hs
- cd model/$MODEL_FOLDER
- hs cluster add --name <name> --server <server_addr>
- hs upload
- ```
-  
+### [Titanic prediction](examples/titanic_xgboost)
+
+### [Mnist Python](examples/python_mnist)
```
Comment (Valenzione) [inline README.md]: invalid path `(examples/mnist_py)` is correct one

File: README.md [hunk 6]
```diff
@@ -1,24 +1,54 @@
...
+   Digit classification model
+
+
+## Tensorflow Runtime:
+[Tensorflow serving documentation](https://hydrosphere.io/serving-docs/latest/tutorials/tensorflow.html)
+### [Mnist Tensorflow](examples/mnist)
```
Comment (Valenzione) [inline README.md]: Invalid path. `(examples/mnist_tf)` is correct one

File: examples/adult/demo/adult_demo.ipynb [hunk 7]
```diff
@@ -95,7 +94,7 @@
    },
    "outputs": [],
    "source": [
-    "channel = grpc.secure_channel(\"dev.k8s.hydrosphere.io\", credentials=ssl_channel_credentials()) \n",
+    "channel = grpc.secure_channel(\"localhost:9090\") \n",
```
Comment (Valenzione) [inline examples/adult/demo/adult_demo.ipynb]: Does not localhost have an insecure channel?
